libp2p/nameresolving/dnsresolver.nim

```diff
@@ -107,10 +109,11 @@ method resolveIp*(
         let resp = await fut
         for answer in resp.answers:
           resolvedAddresses.incl(answer.toString())
-      except ValueError as e:
-        info "Failed to query DNS", address, error=e.msg
-      except TransportOsError as e:
+      except CancelledError as e:
+        raise
```

Comment:
we avoid this form because it relies on `getCurrentException` which is problematic - `raise e`

Reply:
Ah right in meant to do `except CancelledError` without `as e`, is this also an issue?

Reply:
yeah, it's an unfortunate design issue in Nim - best avoid it, it works "sometimes" but it's too much of a mess to remember when, this way we only have one form to debug